lib/matplotlib/tests/test_cycles.py

```diff
@@ -1,140 +1,100 @@
 def test_colorcycle_basic():
     fig, ax = plt.subplots()
     ax.set_prop_cycle(cycler('color', ['r', 'g', 'y']))
-    xs = np.arange(10)
-    ys = 0.25 * xs + 2
-    ax.plot(xs, ys, label='red', lw=4)
-    ys = 0.45 * xs + 3
-    ax.plot(xs, ys, label='green', lw=4)
-    ys = 0.65 * xs + 4
-    ax.plot(xs, ys, label='yellow', lw=4)
-    ys = 0.85 * xs + 5
-    ax.plot(xs, ys, label='red2', lw=4)
-    ax.legend(loc='upper left')
-
-
-@image_comparison(['marker_cycle.png', 'marker_cycle.png'], remove_text=True,
-                  tol={'aarch64': 0.02}.get(platform.machine(), 0.0))
+    for _ in range(4):
+        ax.plot(range(10), range(10))
+    assert [l.get_color() for l in ax.lines] == ['r', 'g', 'y', 'r']
+
+
 def test_marker_cycle():
     fig, ax = plt.subplots()
     ax.set_prop_cycle(cycler('c', ['r', 'g', 'y']) +
                       cycler('marker', ['.', '*', 'x']))
-    xs = np.arange(10)
-    ys = 0.25 * xs + 2
-    ax.plot(xs, ys, label='red dot', lw=4, ms=16)
-    ys = 0.45 * xs + 3
-    ax.plot(xs, ys, label='green star', lw=4, ms=16)
-    ys = 0.65 * xs + 4
-    ax.plot(xs, ys, label='yellow x', lw=4, ms=16)
-    ys = 0.85 * xs + 5
-    ax.plot(xs, ys, label='red2 dot', lw=4, ms=16)
-    ax.legend(loc='upper left')
+    for _ in range(4):
+        ax.plot(range(10), range(10))
+    assert [l.get_color() for l in ax.lines] == ['r', 'g', 'y', 'r']
+    assert [l.get_marker() for l in ax.lines] == ['.', '*', 'x', '.']
 
     fig, ax = plt.subplots()
     # Test keyword arguments, numpy arrays, and generic iterators
     ax.set_prop_cycle(c=np.array(['r', 'g', 'y']),
                       marker=iter(['.', '*', 'x']))
-    xs = np.arange(10)
-    ys = 0.25 * xs + 2
-    ax.plot(xs, ys, label='red dot', lw=4, ms=16)
-    ys = 0.45 * xs + 3
-    ax.plot(xs, ys, label='green star', lw=4, ms=16)
-    ys = 0.65 * xs + 4
-    ax.plot(xs, ys, label='yellow x', lw=4, ms=16)
-    ys = 0.85 * xs + 5
-    ax.plot(xs, ys, label='red2 dot', lw=4, ms=16)
-    ax.legend(loc='upper left')
-
-
-@image_comparison(['lineprop_cycle_basic.png'], remove_text=True,
-                  tol={'aarch64': 0.02}.get(platform.machine(), 0.0))
+    for _ in range(4):
+        ax.plot(range(10), range(10))
+    assert [l.get_color() for l in ax.lines] == ['r', 'g', 'y', 'r']
+    assert [l.get_marker() for l in ax.lines] == ['.', '*', 'x', '.']
+
+
 def test_linestylecycle_basic():
     fig, ax = plt.subplots()
     ax.set_prop_cycle(cycler('ls', ['-', '--', ':']))
-    xs = np.arange(10)
-    ys = 0.25 * xs + 2
-    ax.plot(xs, ys, label='solid', lw=4, color='k')
-    ys = 0.45 * xs + 3
-    ax.plot(xs, ys, label='dashed', lw=4, color='k')
-    ys = 0.65 * xs + 4
-    ax.plot(xs, ys, label='dotted', lw=4, color='k')
-    ys = 0.85 * xs + 5
-    ax.plot(xs, ys, label='solid2', lw=4, color='k')
-    ax.legend(loc='upper left')
-
-
-@image_comparison(['fill_cycle_basic.png'], remove_text=True)
+    for _ in range(4):
```

Comment:
this could probably be a test fixture given it's the same for all of these (and I know, for two lines it might not be worth it)

Reply:
I'm not sure fixturizing two lines really helps readability...

Reply:
I'm thinking more about the code repetition, but 🤷‍♀ am +-0 on it.